bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js

```diff
@@ -0,0 +1,113 @@
+import { getSorterFor } from '../../../../src/react/components/Table';
+
+const FEATUREDATA_DEFAULT_HIDDEN_FIELDS = ['__fid', '__centerX', '__centerY', 'geometry'];
+
+class FeatureDataPluginUIHandler extends StateHandler {
+    constructor (plugin, mapModule, config) {
+        super();
+        this.setState({
+            activeTab: null
+        });
+        this.mapModule = mapModule;
+        this.addStateListener(() => this.updateFlyout());
+    }
+
+    getFeatureDataLayers () {
+        return Oskari.getSandbox()
+            .findAllSelectedMapLayers()
+            .filter(layer => layer.isVisibleOnMap() && layer.hasFeatureData && layer.hasFeatureData());
+    }
+
+    getFeaturesByLayerId (layerId) {
+        const featuresMap = this.mapModule.getVectorFeatures(null, { layers: [layerId] });
+        return featuresMap && featuresMap[layerId] ? featuresMap[layerId].features : null;
+    }
+
+    createColumnSettingsFromFeatures (features) {
+        return Object.keys(features[0].properties)
+            .filter(key => !FEATUREDATA_DEFAULT_HIDDEN_FIELDS.includes(key))
+            .map(key => {
+                return {
+                    align: 'left',
+                    title: key,
+                    dataIndex: key,
+                    sorter: getSorterFor(key)
+                };
+            });
+    }
+
+    createDatasourceFromFeatures (features) {
+        return features.map(feature => {
+            return {
+                key: feature.properties.__fid,
+                ...feature.properties
+            };
+        });
+    }
+
+    createLayerTabs (layerId, layers, features) {
+        const tabs = layers?.map(layer => {
+            return {
+                key: layer.getId(),
+                label: layer.getName(),
+                children: layer.getId() === layerId && !!features && features.length
+                    ? createFeaturedataGrid(this.createColumnSettingsFromFeatures(features), this.createDatasourceFromFeatures(features))
+                    : null
+            };
+        }) || [];
+        return tabs;
+    }
+
+    setActiveTab (layerId) {
+        const featureDataLayers = this.getFeatureDataLayers() || null;
+        const features = layerId ? this.getFeaturesByLayerId(layerId) : null;
+        const tabs = this.createLayerTabs(layerId, featureDataLayers, features);
+        this.updateState({
+            activeTab: layerId,
+            tabs
+        });
+    }
+
+    updateStateAfterMapMoveEvent () {
+        const featureDataLayers = this.getFeatureDataLayers() || [];
+        const layerId = featureDataLayers?.find(layer => layer.getId() === this.state.activeTab) ? this.state.activeTab : featureDataLayers.map(layer => layer.getId())[0] || null;
+        const features = layerId ? this.getFeaturesByLayerId(layerId) : [];
+        const tabs = this.createLayerTabs(layerId, featureDataLayers, features);
+        this.updateState({
+            activeTab: layerId,
+            tabs
```

Comment:
Sama juttu tässä. Eli pääsisi helpommalla jos vain päivittää `activeLayerFeatures : getFeaturesByLayerId(this.state.activeTab)`